/*
 * Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package privacy_sandbox.bidding_auction_servers.bidding_service;

message BuyerCodeFetchConfig {

   // The javascript generateBid script.
   string bidding_js_path = 1;

   // URL endpoint for fetching AdTech code blob, js file.
   string bidding_js_url = 2;

   // URL endpoint to the wasm file. AdTechs can choose to include wasm file which will be combined
   // with the js file before loading into Roma.
   string bidding_wasm_helper_url = 3;

   // Period of how often to fetch AdTech code blob from the URL endpoint.
   int64 url_fetch_period_ms = 4;

   // A time out limit for HttpsFetcherAsyc client to stop executing FetchUrl.
   int64 url_fetch_timeout_ms = 5;

   // Allow buyer debug URL generation.
   bool enable_buyer_debug_url_generation = 6;

   // Allow handling of console.logs from AdTech script execution.
   bool enable_adtech_code_logging = 8;

   // URL endpoint for fetching AdTech code blob, js file for protected app
   // signals.
   string protected_app_signals_bidding_js_url = 9;
}
